Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1966,7 +1966,7 @@ void WebPageProxy::loadRequestWithNavigationShared(Ref<WebProcessProxy>&& proces
         if (!weakProcess->isLaunching() || !url.protocolIsFile())
             weakProcess->send(Messages::WebPage::LoadRequest(WTFMove(loadParameters)), webPageID);
         else
-            weakProcess->send(Messages::WebPage::LoadRequestWaitingForProcessLaunch(WTFMove(loadParameters), weakThis->internals().pageLoadState.resourceDirectoryURL(), weakThis->internals().identifier, true), webPageID);
+            weakProcess->send(Messages::WebPage::LoadRequestWaitingForProcessLaunch(WTFMove(loadParameters), weakThis->internals().pageLoadState.resourceDirectoryURL(), weakThis->identifier(), true), webPageID);
```

Comment:
I know you're not changing these, but `weakThis` and `weakProcess` should both be protected.

Reply:
We are focusing IPC related classes for now so that other part is not touched. This change is to settle many warnings in those focus area, WebPageProxy itself is very untouched in smart pointer context.

Reply:
I don't think Charlie is asking for a lot here and his proposal is correct and would make the code safer, which is what you're working on. Therefore, I suggest we apply his suggestions.

Reply:
Got it. I'm fixing visible part.